coursebuilder/modules/explorer/_static/js/course.js

```diff
@@ -15,18 +15,38 @@ $(document).ready(function documentReady() {
   });
 
   var sidebarToggle = false;
-  $('body').on('click', '#hideCourseSidenav', function () {
+  $("body").on("click", "#hideCourseSidenav", function() {
     sidebarToggle = !sidebarToggle;
-    var sidebarWidth = sidebarToggle ? $('.course-accordion').width() : 0;
+    var sidebarWidth = sidebarToggle ? $(".course-accordion").width() : 0;
 
-    $('.course-accordion').toggleClass('hide-sidenav');
-    $('.course-card').animate({'margin-left': sidebarWidth / 2 + 'px'}, 300);
-    $('.course-accordion').animate({'margin-left': '-' + sidebarWidth + 'px'}, 300);
+    $(".course-accordion").toggleClass("hide-sidenav");
+    $(".course-card").animate({ "margin-left": sidebarWidth / 2 + "px" }, 300);
+    $(".course-accordion").animate(
+      { "margin-left": "-" + sidebarWidth + "px" },
+      300
+    );
   });
 
-  $('.course-accordion .collapsible').clone().appendTo('#courseMobileSidebar');
-  $('.collapsible').collapsible();
-  $('body').on('click', '#closeSideNav', function () {
-    $('.button-collapse').sideNav('hide');
+  $(".course-accordion .collapsible").clone().appendTo("#courseMobileSidebar");
+  $(".collapsible").collapsible();
+  $("body").on("click", "#closeSideNav", function() {
+    $(".button-collapse").sideNav("hide");
+  });
+
+  $(".course-carousel").on("click", function(event) {
+    $(document.documentElement).off("keyup");
+    var owlCarousel = $(event.target).closest(".owl-carousel");
+    var target = $(event.target);
+    $(document.documentElement).on("keyup", function (event) {
+      if (target.closest(".owl-stage-outer").length) {
+        event.stopPropagation();
```

Comment:
@andela-johia When do you want to stopPropagation() and is it for all keyup events?

Reply:
@chegejames event.stopPropagation() is invoked when the  keyup event occurs inside the `owl-stage-outer` div so the event doesn't bubble up the DOM into other parent and grand-parents element. Yes it is for all keyup events.